lib/SILOptimizer/Mandatory/ClosureLifetimeFixup.cpp

```diff
@@ -1431,6 +1482,11 @@ static bool fixupClosureLifetimes(SILFunction &fn,
       if (!cvt || cvt->isLifetimeGuaranteed())
         continue;
 
+      if (tryHandleConvertOfDifferentiableFunction(cvt)) {
+        changed = true;
```

Comment:
Why do we need to set `changed` to `true` here?

Reply:
I guess it's setting `changed` because it does `convertEscapeToNoEscapeInst->setLifetimeGuaranteed()`.
But the changed value is not true under the same condition as the transformation. Why is the transformation to a guaranteed lifetime happening unconditionally for all `ThinToThickFunctionInst`s?

Reply:
> But the changed value is not true under the same condition as the transformation.

The changed value tells us whether one or more closures' lifetimes were "fixed up", no? 

> Why is the transformation to a guaranteed lifetime happening unconditionally for all ThinToThickFunctionInsts?

I was surprised that we don't do this for other `ThinToThickFunctionInst`s, but I wasn't sure if we could have a sequence where a `ThinToThickFunctionInst` is later partially applied (outside of thunks), like this:
```swift
%tttf = thin_to_thick_func %1: $@convention(thin) (Float, Float) -> Float
... // one or more conversions and/or reabstractions
%pai = partial_apply %modified_tttf(%some_float):  @callee_guaranteed (Float, Float) -> Float
```
At least for Autodiff we should never have such a pattern (or if we do, the logic in this PR will just skip it).

Is it not right to simply set the lifetimes of closures derived from `ThinToThickFunctionInst`s as guaranteed, in this case?